.github/actions/trun/run.ts

```diff
@@ -0,0 +1,132 @@
+import * as esbuild from "./deps/esbuild.ts"
+import { denoPlugin } from "./deps/esbuild_deno_loader.ts"
+import PQueue from "./deps/pqueue.ts"
+import { Browser } from "./deps/puppeteer.ts"
+import { deferred } from "./deps/std/async.ts"
+import { Buffer, readLines } from "./deps/std/io.ts"
+import * as path from "./deps/std/path.ts"
+import { readerFromStreamReader, writeAll } from "./deps/std/streams.ts"
+
+export interface RunOptions {
+  paths: (readonly [dir: string, fileName: string])[]
+  concurrency: number
+  runner: (dir: string, fileName: string) => Promise<void>
+}
+
+export async function run({ paths, concurrency, runner }: RunOptions) {
+  const runQueue = new PQueue({ concurrency })
+  runQueue.addAll(paths.map(([dir, fileName]) => () => runner(dir, fileName)))
+  await runQueue.onIdle()
+}
+
+export interface RunWithBrowserOptions {
+  createBrowser: () => Promise<Browser>
+  importMapURL?: URL
+  results: [fileName: string, exitCode: number][]
+}
+
+export async function runWithBrowser(
+  { createBrowser, importMapURL, results }: RunWithBrowserOptions,
+) {
+  const browser = await createBrowser()
+  const currentDir = new URL(import.meta.url).pathname.split("/").slice(0, -1).join("/")
+  const consoleJS = new TextDecoder().decode(await Deno.readFile(`${currentDir}/console.js`))
+
+  return (async (dir: string, fileName: string) => {
+    console.log(`running ${fileName}`)
+    const outputQueue = new PQueue({ concurrency: 1, autoStart: false })
+
+    outputQueue.add(() => console.log(`${fileName} output:`))
+
+    const page = await browser.newPage()
+    const result = await esbuild.build({
+      plugins: [
+        denoPlugin({
+          importMapURL,
+        }) as any,
+      ],
+      entryPoints: [path.join(dir, fileName)],
+      bundle: true,
+      write: false,
+      format: "esm",
+    })
```

Comment:
It would be nice to run this using the `dnt` build.
That way we would be testing that the packaged build works in the browser too.